WooCommerce/Classes/ViewRelated/Products/ShareProduct/ShareProductCoordinator.swift

```diff
@@ -0,0 +1,102 @@
+import UIKit
+import struct Yosemite.Site
+import protocol Experiments.FeatureFlagService
+
+/// Coordinates navigation for product sharing
+final class ShareProductCoordinator: Coordinator {
+    let navigationController: UINavigationController
+
+    private let site: Site?
+    private let productURL: URL
+    private let productName: String
+    private let shareSheetAnchorView: UIView?
+    private let shareSheetAnchorItem: UIBarButtonItem?
+    private let viewControllerToPresentShareSheet: UIViewController
```

Comment:
❓ Is this controller necessary? IMO we can present the sheet from the navigation controller, or from the top view controller of the stack.

Reply:
Good idea. Done in https://github.com/woocommerce/woocommerce-ios/pull/9875/commits/0e04cfb2c8baf118a824818ae79082e7ea8c78d5 and https://github.com/woocommerce/woocommerce-ios/pull/9875/commits/3c2943a49f0d4f681b27cd9fe2729c9b00160f0b